service/service.go

```diff
@@ -472,7 +472,55 @@ func (s *Service) ListEvents(ctx context.Context, req *api.ListEventsRequest) (*
 		}
 		events, err = s.store.GetEventsByExecutionIDs([]mlmetadata.ExecutionID{mlmetadata.ExecutionID(id)})
 	}
+	artifacts, err := s.getArtifactsInEvents(events)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get artifacts in events: %v", err)
+	}
+	executions, err := s.getExecutionsInEvents(events)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get executions in events: %v", err)
+	}
 	return &api.ListEventsResponse{
-		Events: events,
+		Events:     events,
+		Artifacts:  artifacts,
+		Executions: executions,
 	}, err
 }
+
+func (s *Service) getArtifactsInEvents(events []*mlpb.Event) (map[int64]*mlpb.Artifact, error) {
+	results := make(map[int64]*mlpb.Artifact)
+	for _, e := range events {
+		id := e.GetArtifactId()
+		if _, exists := results[id]; exists {
```

Comment:
nit: `exists` => `ok` as that's the common way to write this ('comma ok idiom'). Feel free to ignore :-)

Reply:
I will use this idiom in future PRs :-) Thanks